searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp

```diff
@@ -416,9 +416,21 @@ class DirectWandBlueprint : public queryeval::ComplexLeafBlueprint
                         .setDocIdLimit(get_docid_limit()),
                 _weights, _terms, _attr, strict);
     }
+    std::unique_ptr<SearchIterator> createFilterSearch(bool strict, FilterConstraint constraint) const override;
     bool always_needs_unpack() const override { return true; }
 };
 
+std::unique_ptr<SearchIterator>
+DirectWandBlueprint::createFilterSearch(bool, FilterConstraint) const
+{
+    std::vector<DocumentWeightIterator> iterators;
+    iterators.reserve(_terms.size());
+    for (const IDocumentWeightAttribute::LookupResult &r : _terms) {
+        _attr.create(r.posting_idx, iterators);
+    }
+    return attribute::DocumentWeightOrFilterSearch::create(std::move(iterators));
```

Comment:
when constraint is LOWER_BOUND, all WeakAnd variants must create filter as if it's AND.

Reply:
I'll change it to use EmptySearch for LOWER_BOUND.

Reply:
Fixed.